### PR TITLE
SYSE-354/add-test-controller-exclusions

### PR DESCRIPTION
**Pull request by ermirizio**
- Adds the exclusion statements to the test-controller in order to avoid undesired combinations

### CODE REVIEW

File: policy/controller.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ func sortedKeys[K constraints.Ordered, V any](m map[K]V) []K {
 
 // TestVariations models the variations of the test matrix in
 // release.yml:api-tests. Each key is a row in the matrix.
-type TestVariations map[string][]string
+type TestVariations map[string]interface{}
```
Comment (alephnull) [inline policy/controller.go:33]: map[string]interface invalidates the type guarantees that we need make sure that the correct type is passed to the matrix.
Perhaps the excludes can be a new type?

**Review by alephnull — CHANGES_REQUESTED** [review]

File: policy/controller.go (hunk 1 above)
Comment (ermirizio) [reply policy/controller.go:33]: yes in that case we need a new type, I was between that or using interface. Maybe we can build a struct that contains the 2 types so it's easier to pass between functions or call methods?

**Comment by ermirizio** [comment]
@alephnull let me know what do you think. If this still requires changes I would prefer to do a pairing session or you can suggest changes if prefer to speed up the process.

File: cmd/cmdtest/policy/commands.ct [hunk 2]
```diff
@@ -28,3 +28,6 @@ EOF
 sink<<EOF
 ["$ECR/tyk-sink:master"]
 EOF
+exclude<<EOF
+[]
```
Comment (alephnull) [inline cmd/cmdtest/policy/commands.ct:32]: The code in defaults below has exclusions, so how come this array is empty?

File: policy/controller.go [hunk 3]
```diff
@@ -15,9 +15,9 @@ import (
 	"golang.org/x/exp/constraints"
 )
 
-// policy.SetVariations needs to print the variations in a
+// policy.SetOutputs needs to print the variations in a
 // well-defined order for TestOutput
-func sortedKeys[K constraints.Ordered, V any](m map[K]V) []K {
+func sortedVariations[K constraints.Ordered, V any](m map[K]V) []K {
```
Comment (alephnull) [inline policy/controller.go:20]: This is a generic function that can operate on any ordered type. Renaming it is not required.

**Review by alephnull — APPROVED** [review]
Largely looks good. Some minor points.

File: cmd/cmdtest/policy/commands.ct (hunk 2 above)
Comment (ermirizio) [reply cmd/cmdtest/policy/commands.ct:32]: $ setenv IS_PR yes is not default given the case is within is_pr trigger:

image

**Pull request merged**